lisa/analysis/pixel6.py

```diff
@@ -0,0 +1,89 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2023, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pandas as pd
+import holoviews as hv
+from holoviews import opts
+
+from lisa.datautils import df_add_delta
+from lisa.analysis.base import TraceAnalysisBase
+from lisa.trace import requires_events
+from lisa.notebook import plot_signal
+
+
+class Pixel6Analysis(TraceAnalysisBase):
+    """
+    Support for Pixel 6-specific analysis data
+
+    :param trace: input Trace object
+    :type trace: :class:`lisa.trace.Trace`
+    """
+
+    name = 'pixel6'
+
+    EMETER_CHAN_NAMES = {
+        'S4M_VDD_CPUCL0': 'little',
+        'S3M_VDD_CPUCL1': 'mid',
+        'S2M_VDD_CPUCL2': 'big',
+    }
+
+###############################################################################
+# DataFrame Getter Methods
+###############################################################################
+    @TraceAnalysisBase.cache
+    @requires_events('pixel6_emeter')
+    def df_power_meter(self):
+        """
+        Get the power meter readings across the trace.
+
+        :retuns: A :class:`pandas.DataFrame` with:
+            * A ``channel`` column (description of the power usage channel)
+            * A ``value`` column (average power usage in mW since the last measurement)
+        """
+        df = self.trace.df_event('pixel6_emeter')
+        channels = list(Pixel6Analysis.EMETER_CHAN_NAMES.keys())
+        df = df.groupby(['chan_name'], observed=True).apply(
+            lambda x: df_add_delta(x, col='value_diff', src_col='value', window=self.trace.window)['value_diff'] / df_add_delta(x, col='ts_diff', src_col='ts', window=self.trace.window)['ts_diff']
+        ).reset_index().query("chan_name in @channels").rename(columns={0: 'value', 'chan_name': 'channel'}).dropna().set_index('Time')
+        df['channel'] = df['channel'].apply(lambda s: Pixel6Analysis.EMETER_CHAN_NAMES[s])
```

Comment:
this:
```
.apply(lambda s: Pixel6Analysis.EMETER_CHAN_NAMES[s])
```

can be replaced by:
```
.map(Pixel6Analysis.EMETER_CHAN_NAMES)
```

or even (much) better yet, change the column dtype to `category` (df['foo'] = df['foo'].astype('category')) and modify the category with the new names. This will avoid having to lookup the value of each row in the dict